desc/coils.py

```diff
@@ -1172,18 +1185,14 @@ def from_symmetry(cls, coils, NFP=1, sym=False):
         # check toroidal extent of coils to be repeated
         maxphi = 2 * np.pi / NFP / (sym + 1)
         data = coils.compute("phi")
-        for i, cdata in enumerate(data):
-            errorif(
-                np.any(cdata["phi"] > maxphi),
-                ValueError,
-                f"coil {i} exceeds the toroidal extent for NFP={NFP} and sym={sym}",
-            )
-            warnif(
-                sym and np.any(cdata["phi"] < np.finfo(cdata["phi"].dtype).eps),
-                UserWarning,
-                f"coil {i} is on the symmetry plane phi=0",
-            )
-
+        check_for_intersection = False
+        for cdata in data:
+            # check if any coils lie on symmetry planes, these may cause
+            # self-intersection after reflection/rotation
+            check_for_intersection = (
+                np.any(cdata["phi"] > maxphi)
+                or (sym and np.any(cdata["phi"] < np.finfo(cdata["phi"].dtype).eps))
+            ) or check_for_intersection
```

Comment:
If you want to get fancy (and I think is more readable code): 
```
check_intersection = any(
    [
        np.any(cdata["phi"] > maxphi)
        or (sym and np.any(cdata["phi"] < np.finfo(cdata["phi"].dtype).eps))
        for cdata in data
    ]
)
```

Reply:
See comment below. We probably don't actually need this check at all if we always check for intersections in `CoilSet.__init__` by default anyways.